mlc_llm/relax_model/llama.py

```diff
@@ -800,12 +877,55 @@ def get_model(args, hf_config):
                 },
             )
 
+    def f_convert_pname_fwd(pname: str) -> List[str]:
+        if not config.combine_matmul:
+            return [pname]
+
+        qkv_str = "query_key_value_proj"
+        gate_up_str = "gate_up_proj"
+        if qkv_str in pname:
+            return [
+                pname.replace(qkv_str, "q_proj"),
+                pname.replace(qkv_str, "k_proj"),
+                pname.replace(qkv_str, "v_proj"),
+            ]
+        elif gate_up_str in pname:
+            return [
+                pname.replace(gate_up_str, "gate_proj"),
+                pname.replace(gate_up_str, "up_proj"),
+            ]
+        else:
+            return [pname]
+
+    def f_convert_param_bkwd(torch_pname: str, torch_param):
+        if not config.combine_matmul:
+            return [(torch_pname, torch_param.astype(dtype))]
+
+        combined_layers = ["q_proj", "k_proj", "v_proj", "gate_proj", "up_proj"]
+        if any([name in torch_pname for name in combined_layers]):
+            return None
+        return [(torch_pname, torch_param.astype(dtype))]
+
+    def f_compute_relax_param(relax_pname: str, torch_params: List[Any]):
+        if not config.combine_matmul:
+            raise NotImplementedError()
```

Comment:
Better add a short explanation here.

Reply:
Added. Does it look good now?

Reply:
I mean some explaination inside NotImplementedErrer :)

Reply:
Got it. Done now. We also have some docstring now
https://github.com/mlc-ai/mlc-llm/blob/e36a38d75c87ee184be2ae20b696069b376a25a8/mlc_llm/relax_model/param_manager.py#L134-L143